app/src/main/java/de/tum/in/tumcampusapp/component/ui/search/adapter/ResultTypesAdapter.kt

```diff
@@ -42,49 +39,65 @@ class ResultTypesAdapter(
 
         fun bind(typeData: ResultTypeData) {
             currentType = typeData
-            val text = when (typeData.type) {
+            val textId = when (typeData.type) {
                 SearchResultType.ALL -> R.string.all_results
                 SearchResultType.PERSON -> R.string.people
                 SearchResultType.LECTURE -> R.string.lectures
                 SearchResultType.NAVIGA_ROOM -> R.string.rooms_id
                 SearchResultType.BUILDING -> R.string.buildings
             }
-            typeTextView.setText(text)
 
-            if (typeData.type == typeData.selectedType) {
-                typeTextView.textColorResource = R.color.white
-                layout.setBackgroundResource(R.drawable.search_result_selected_type_background)
+            chip.text = chip.context.resources.getText(textId)
+            setIsSelected(typeData.selected)
+
+            if (typeData.type == SearchResultType.ALL) {
+                itemView.margin(left = 16F)
             } else {
-                typeTextView.textColorResource = R.color.text_primary
-                layout.setBackgroundResource(R.drawable.search_result_type_background)
+                itemView.margin(left = 8F)
             }
+        }
 
-            if (typeData.type == SearchResultType.ALL) {
-                layout.margin(left = 16F)
+        fun setIsSelected(selected: Boolean) {
+            if (selected) {
+                chip.isChecked = true
+                chip.isCheckable = false
             } else {
-                layout.margin(left = 8F)
+                chip.isCheckable = true
+                chip.isChecked = false
             }
         }
     }
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ResultTypeViewHolder {
         val view = LayoutInflater.from(parent.context)
-            .inflate(R.layout.search_result_type, parent, false)
+            .inflate(R.layout.search_result_chip, parent, false)
         return ResultTypeViewHolder(view, onClick)
     }
 
     override fun onBindViewHolder(holder: ResultTypeViewHolder, position: Int) {
         val resultType = getItem(position)
         holder.bind(resultType)
     }
+
+    override fun onBindViewHolder(holder: ResultTypeViewHolder, position: Int, payloads: MutableList<Any>) {
+        if (payloads.isEmpty()) {
+            super.onBindViewHolder(holder, position, payloads)
+        } else {
+            holder.setIsSelected(payloads[0] as Boolean)
+        }
+    }
```

Comment:
I dont get this part: Could you explain this (What is this doing?)

Reply:
Because I have overridden `getChangePayload()`, whenever new data is submitted and the only difference between old and new buttons of the same name is the "selected"-state, the new state is added as a payload. In these cases it is not necessary to replace the existing view using `super.onBindViewHolder()` as the existing one can be updated using `setIsSelected()` instead.

Reply:
I think adding a comment explaining this circumstance could be worthwhile

Reply:
I've added two comments which should explain the idea.